src/content/posts/2023-2-terraform-vs-cdk-vs-cloudformation.md

```diff
@@ -0,0 +1,574 @@
+  * **Security and Governance** - it’s easier to ensure security standards are met with IaC. You can use tools such as tfsec or AWS CloudFormation Guard and make them run even before the infrastructure is created.
+
+## Tools comparison
+
+Let's compare the most commonly used infrastructure as code tools:
+* [Terraform](https://www.terraform.io/)
+* [AWS CloudFormation](https://aws.amazon.com/cloudformation/faqs/) (AWS CFN)
+* [AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html)
+
+While the above tools fullfil the same purpose (allow us to treat and manage infrastructure as code), they are vastly different in how they work, their workflows, integrations, code logic etc. It's important to note that while Terraform and AWS CloudFormation are two separate tools, **AWS CDK is built on top of AWS CloudFormation**. This means, that when you use AWS CDK, you also use AWS CloudFormation. AWS CDK just adds another layer to the tool. Read more here about how [AWS CDK is powered by AWS CloudFormation](https://aws.amazon.com/cdk/features/)
+
+Below is a table detailing the main differences between the IaC tools.
+
+### Main differences
+| Feature | AWS CloudFormation | AWS CDK | Terraform |
+|----------|-----------|----------| -- |
+| Language | YAML or JSON or AWS CloudFormation Designer - a GUI | TypeScript, Python, Java, .NET, or Go (in Developer Preview). | Hashicorp (HCL) syntax |
+| Supported infrastructure  | AWS only (plus 3rd party modules to support e.g. monitoring or incident management; [link](https://aws.amazon.com/cloudformation/features/?pg=ln&sec=hs#extensibility))  | Same as AWS CloudFormation | Various clouds and PaaS integrations (e.g. AWS, Azure, GCP, Kubernetes, Alibaba Cloud. Find out about other Terraform providers [here](https://registry.terraform.io/browse/providers)) |
+| State management | Handled for you | Same as AWS CloudFormation | Needs to be configured by you (many options) |
+| Version management | You cannot choose which version of AWS CFN you're using (the history of changes is available [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/ReleaseHistory.html)) | Same as AWS CloudFormation | You can choose which version of Terraform and each Terraform provider you use (the history of changes is available [here](https://github.com/hashicorp/terraform/blob/main/CHANGELOG.md)) |
+| Pricing | [Not free](https://aws.amazon.com/cloudformation/pricing/), but also subjectively not expensive | Same as AWS CloudFormation | Fee |
+| License | It's one of many AWS services | Open-source | Until 11.08.2023 Terraform used be open-source, today it's under the BSL license, which really only impacts the Terraform competitors. Read more [here](https://kudulab.io/posts/2023-1-terraform-license-change/) |
+
+This leads to the following conclusions:
+* if you want to deploy infrastructure that is not AWS, go with Terraform
+* if you want to use only AWS-provided tools, go with AWS CloudFormation or AWS CDK
```

Comment:
if you want to use only AWS infrastructure, you could consider AWS CloudFormation or AWS CDK